### PR TITLE
Introduce config sub-state and reducer

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

File: packages/core/src/reducers/config.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as _ from 'lodash';
+import { SET_CONFIG } from '../actions';
+import { configDefault } from '../configDefault';
+
+const applyDefaultConfiguration = (config?: any) => {
```
Comment (eneufeld) [inline packages/core/src/reducers/config.ts]: can't we default the config to an empty object and always merge?

File: packages/examples/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  'en-US': {
+    postalcode: 'Postal Code'
+  },
+  'de-DE': {
+    postalcode: 'Postleitzahl'
+  }
+};
+
+const config = {
+  restrict: false,
```
Comment (eneufeld) [inline packages/examples/src/config.ts]: we should use non default values to show off the config

Comment (edgarmueller) [reply packages/examples/src/config.ts]: should be both `true`

File: packages/core/src/util/renderer.ts [hunk 3]
```diff
@@ -139,6 +140,7 @@ export const mapStateToControlProps = (state, ownProps): StatePropsOfControl =>
       controlElement.scope !== undefined && isRequired(ownProps.schema, controlElement.scope);
   const resolvedSchema = Resolve.schema(ownProps.schema, controlElement.scope);
   const description = resolvedSchema !== undefined ? resolvedSchema.description : '';
+  const config = getConfig(state);
```
Comment (eneufeld) [inline packages/core/src/util/renderer.ts]: didn't we want to merge the config of the state with the options of the uischema?

File: packages/core/src/util/field.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export const mapStateToFieldProps = (state, ownProps): StatePropsOfField => {
   const controlElement = ownProps.uischema as ControlElement;
   const id = controlElement.scope || '';
   const inputClassName = ['validate'].concat(isValid ? 'valid' : 'invalid');
+  const config = getConfig(state);
```
Comment (eneufeld) [inline packages/core/src/util/field.ts]: didn't we want to merge the config of the state with the options of the uischema?

**Pull request merged**